### PR TITLE
Make energy net use long

### PR DESCRIPTION
**Pull request by Mooy1**
## Description
This will allow energy networks to have a total production/capactity of more than 2.1 billion. Previously it would reset to 0 or go negative.

image

## Proposed changes
Changed a fews ints to longs

## Related Issues (if applicable)
x

## Checklist

- [ ] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [ ] I have also tested the proposed changes in combination with various popular addons and can confirm my changes do not break them.
- [ ] I followed the existing code standards and didn't mess up the formatting.
- [ ] I did my best to add documentation to any public classes or methods I added.
- [ ] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values
- [ ] I added sufficient Unit Tests to cover my code.

### CODE REVIEW

**Comment by WalshyDev** [comment]
This is delaying the problem rather than solving it. It's also causing new issues since you cast to int in other places.

This should instead just be Math.max'd.

**Comment by TheBusyBiscuit** [comment]
Agreed, this just delays the underlying issue.
Also I think the energy net limit should maybe become a config option.

**Comment by Mooy1** [comment]
Why a config option? No other energy related stuff is in configs.

**Comment by WalshyDev** [comment]
Configuration is good and it's very easy to just add here. Also allows for users to make it harder and limit it to like 10k.

**Comment by Mooy1** [comment]
> This is delaying the problem rather than solving it. It's also causing new issues since you cast to int in other places.
> 
> This should instead just be Math.max'd.

Casting to int is fine since anywhere that its cast is for sure under Integer.MAX_VALUE, and what should i Math.max?

File: src/main/java/io/github/thebusybiscuit/slimefun4/core/networks/energy/EnergyNet.java [hunk 1]
```diff
@@ -231,7 +227,15 @@ private int tickAllGenerators(@Nonnull LongConsumer timings) {
...
-                    supply += energy;
+                    supply += provider.getGeneratedOutput(loc, data);
+
+                    if (provider.isChargeable()) {
+                        supply += provider.getCharge(loc, data);
+                    }
+                    
+                    if (supply < 0 || supply > MAX_ENERGY) {
+                        supply = MAX_ENERGY;
+                    }
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/core/networks/energy/EnergyNet.java:238]: Congratulations, you broke reactors.

**Review by TheBusyBiscuit — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/thebusybiscuit/slimefun4/core/networks/energy/EnergyNet.java (hunk 1 above)
Comment (Mooy1) [reply src/main/java/io/github/thebusybiscuit/slimefun4/core/networks/energy/EnergyNet.java:238]: forgot that getGeneratedOutput ran other code xD

**Review by TheBusyBiscuit — CHANGES_REQUESTED** [review]
I don't know what @WalshyDev might think about this.
But I would be in favour of discarding this pull request for now.

This is no simple change that has massive consequences for the entire codebase and merging something like this without giving it proper thought would have severe side effects for the whole project.
Thanks for bringing this up but I would be against merging it as of now, but we may revisit this at a later time, though it should probably be done properly on the entire energy code then.

**Comment by TheBusyBiscuit** [comment]
Closed, as discussed.

**Pull request closed without merging**